contract/models/contract.py

```diff
@@ -0,0 +1,470 @@
+# Copyright 2004-2010 OpenERP SA
+# Copyright 2014 Angel Moya <angel.moya@domatix.com>
+# Copyright 2015 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2016-2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# Copyright 2016-2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+
+
+class ContractContract(models.Model):
+    _name = 'contract.contract'
+    _inherit = ['mail.thread', 'contract.abstract.contract']
+
+    active = fields.Boolean(
+        default=True,
+    )
+    code = fields.Char(
+        string="Reference",
+    )
+    group_id = fields.Many2one(
+        string="Group",
+        comodel_name='account.analytic.account',
+        ondelete='restrict',
+    )
+    analytic_account_id = fields.Many2one(
+        string="Analytic account",
+        comodel_name='account.analytic.account',
+    )
+    currency_id = fields.Many2one(
+        related="company_id.currency_id",
+        string="Currency",
+        readonly=True,
+    )
+    contract_template_id = fields.Many2one(
+        string='Contract Template', comodel_name='contract.template'
+    )
+    contract_line_ids = fields.One2many(
+        string='Contract lines',
+        comodel_name='contract.line',
+        inverse_name='contract_id',
+        copy=True,
+        oldnae='contract_line_ids',
+    )
+    recurring_invoices = fields.Boolean(
+        string='Generate recurring invoices automatically'
+    )
+    user_id = fields.Many2one(
+        comodel_name='res.users',
+        string='Responsible',
+        index=True,
+        default=lambda self: self.env.user,
+    )
+    create_invoice_visibility = fields.Boolean(
+        compute='_compute_create_invoice_visibility'
+    )
+    recurring_next_date = fields.Date(
+        compute='_compute_recurring_next_date',
+        string='Date of Next Invoice',
+        store=True,
+    )
+    date_end = fields.Date(
+        compute='_compute_date_end', string='Date End', store=True
+    )
+    payment_term_id = fields.Many2one(
+        comodel_name='account.payment.term', string='Payment Terms', index=True
+    )
+    invoice_count = fields.Integer(compute="_compute_invoice_count")
+    fiscal_position_id = fields.Many2one(
+        comodel_name='account.fiscal.position',
+        string='Fiscal Position',
+        ondelete='restrict',
+    )
+    invoice_partner_id = fields.Many2one(
+        string="Invoicing contact",
+        comodel_name='res.partner',
+        ondelete='restrict',
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner', inverse='_inverse_partner_id'
```

Comment:
This is not a computed field. Why do you use an inverse?

Reply:
A field doesn't have to be computed to use an inverse on it. ping @sbejaoui

Reply:
Isn't this weird? For what is this intended?

Reply:
IMO, we can remove the inverse method as `_inverse_partner_id` code is redundant with `_onchange_partner_id`

Reply:
Better to keep inverse method, and on onchange, call to the inverse method. This way, you cover both programatical and UI update.